api/v3/Order.php

```diff
@@ -80,12 +80,14 @@ function _civicrm_api3_order_get_spec(&$params) {
  * @param array $params
  *   Input parameters.
  *
- * @throws API_Exception
  * @return array
  *   Api result array
+ *
+ * @throws \CiviCRM_API3_Exception
+ * @throws API_Exception
  */
 function civicrm_api3_order_create($params) {
-
+  civicrm_api3_verify_one_mandatory($params, NULL, ['line_items', 'total_amount']);
```

Comment:
No, the 'total_amount' should not be required but be optional. If it is present, check that the line items total to it. If it is absent, set the total_amount to the sum of the line item amounts.

Reply:
Right so EITHER line_items OR total_amount are mandatory with this line